lab_09_experiments.py

```diff
@@ -0,0 +1,188 @@
+            "border_color": None,
+            "bg_color": arcade.color.GREEN,
+            "bg_color_pressed": arcade.color.WHITE,
+            "border_color_pressed": arcade.color.WHITE,
+            "font_color_pressed": arcade.color.BLACK,
+        }
+
+        start_button = arcade.gui.UIFlatButton(
+            text="Game", width=200, height=100, style=button_text_style
+        )
+        start_button.on_click = self.hide
+
+        quit_button = arcade.gui.UIFlatButton(
+            text="Exit Game", width=200, height=100, style=button_text_style
+        )
+        quit_button.on_click = lambda event: exit()
+
+        v_box = arcade.gui.UIBoxLayout()
+        v_box.add(start_button.with_space_around(bottom=20))
+        v_box.add(quit_button.with_space_around(bottom=20))
+
+        self.uimanager.add(
+            arcade.gui.UIAnchorWidget(
+                anchor_x="center_x", anchor_y="center_y", child=v_box
+            )
+        )
+        self.show()
+
+    def draw(self):
+        if self.uimanager._enabled:
+            self.uimanager.draw()
+
+    def show(self):
+        self.uimanager.enable()
+        self.shown = True
+
+    def hide(self, event):
+        self.uimanager.disable()
+        self.shown = False
+
+
+class MainGame(arcade.Window):
+    def __init__(self):
+        super().__init__(1000, 1000, title="My Super Game", resizable=True)
+        arcade.set_background_color(arcade.color.YELLOW)
+        self.scene = None
+        self.menu = Menu()
+
+    def setup(self):
+        NUMBER_OF_WORMS = 10
+
+        self.scene = arcade.Scene()
+
+        self.player_sprite = Player("Mole.png", 0.25, center_x=500, center_y=100)
+        self.scene.add_sprite("Player", self.player_sprite)
+
+        self.attack = arcade.load_animated_gif("Attack.gif")
+
+        self.worms = arcade.SpriteList()
+        for i in range(NUMBER_OF_WORMS):
+            self.worms.append(Worm())
+
+        self.scene.add_sprite_list(f"Worms", sprite_list=self.worms)
+
+    def on_draw(self):
+        self.clear()
+        self.scene.draw()
+        self.menu.draw()
+
+    def update(self, deltatime):
+        if not self.menu.shown:
+            self.scene.update()
+
+            if self.attack.sprite_lists:
+                # sometimes fails with TypeError: '<=' not supported between instances of 'NoneType' and 'float'!
+                worm_hit_list = arcade.check_for_collision_with_list(
+                    self.attack, self.worms
+                )
+                for worm in worm_hit_list:
+                    worm.kill()
+
+                self.attack.update_animation(delta_time=1 / 100)
+                if self.attack.cur_frame_idx > 8:
+                    self.attack.cur_frame_idx = 0
+                    self.attack.kill()
+
+    def on_key_press(self, key, modifiers):
+        if self.menu.shown:
+            return
+
+        if key == arcade.key.ESCAPE:
+            self.menu.show()
+
+        if key == arcade.key.W:
```

Comment:
How about creating a set of methods and using them here? E.g. `self.player.move_up()`